components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/device.js

```diff
@@ -383,86 +328,21 @@ deviceModule = function () {
-        for (var g in user_groups) {
-            var deviceInGroup = user_groups[g].devices;
-            deviceCount += deviceInGroup.length;
-            if (deviceInGroup && deviceInGroup.length == 0) {
-                delete user_groups[g]["devices"];
-            }
-            var device;
-            for (var d in deviceInGroup){
-                device = deviceInGroup[d];
-                device.assetId = publicMethods.getAssetId(device.type);
-            }
-            allDevices.push(user_groups[g]);
-        }
-        allDevices.push({id: 0, devices: unGroupedDevices});
-        result.data = allDevices;
-        result.device_count = deviceCount;
-        return result;
-    };
-
-    publicMethods.removeDevice = function (deviceType, deviceId) {
-        var carbonUser = session.get(constants.USER_SESSION_KEY);
-        if (!carbonUser) {
-            log.error("User object was not found in the session");
-            throw constants.ERRORS.USER_NOT_FOUND;
-        }
-        try {
-            utility.startTenantFlow(carbonUser);
-            var deviceManagementService = utility.getDeviceManagementService();
-            var deviceIdentifier = new DeviceIdentifier();
-            deviceIdentifier.setType(deviceType);
-            deviceIdentifier.setId(deviceId);
-            return deviceManagementService.disenrollDevice(deviceIdentifier);
-        } catch (e) {
-            throw e;
-        } finally {
-            utility.endTenantFlow();
-        }
-    };
-
-    publicMethods.updateDevice = function (deviceType, deviceId, deviceName) {
-        var carbonUser = session.get(constants.USER_SESSION_KEY);
-        if (!carbonUser) {
-            log.error("User object was not found in the session");
-            throw constants.ERRORS.USER_NOT_FOUND;
-        }
-        try {
-            utility.startTenantFlow(carbonUser);
-            var deviceManagementService = utility.getDeviceManagementService();
-            var deviceIdentifier = new DeviceIdentifier();
-            deviceIdentifier.setType(deviceType);
-            deviceIdentifier.setId(deviceId);
-            var device = deviceManagementService.getDevice(deviceIdentifier);
-            device.setName(deviceName);
-            return deviceManagementService.modifyEnrollment(device);
-        } catch (e) {
-            throw e;
-        } finally {
-            utility.endTenantFlow();
-        }
+    publicMethods.getDevices = function (userName) {
+        var url = devicemgtProps["httpsURL"] + constants.ADMIN_SERVICE_CONTEXT + "/devices/user/" + userName;
+        return serviceInvokers.XMLHttp.get(
+                url, function (responsePayload) {
+                    for (var i = 0; i < responsePayload.length; i++) {
+                        responsePayload[i].thumb = utility.getDeviceThumb(responsePayload[i].type);
+                    }
+                    return responsePayload;
+                }
+                ,
```

Comment:
Acknowledged. Fixed in commit https://github.com/charithag/carbon-device-mgt/commit/15a6e4b63b6c0fc6de53567f76ff471b3b9010ed